src/main/scala/diplomacy/Nodes.scala

```diff
@@ -355,7 +355,11 @@ sealed abstract class MixedNode[DI, UI, EI, BI <: Data, DO, UO, EO, BO <: Data](
       if (oParamsCycleGuard) throw DownwardCycleException()
       oParamsCycleGuard = true
       val o = mapParamsD(oPorts.size, diParams)
-      require (o.size == oPorts.size, s"Diplomacy error: $context has ${o.size} != ${oPorts.size} down/up outer parameters")
+      require (o.size == oPorts.size,
+        s"""Diplomacy error: $context has ${o.size} != ${oPorts.size} down/up outer parameters:
+           |down-out parameters: [${o.mkString(",")}]
+           |oPorts: [${oPorts.map(_._2.name).mkString(",")}]
```

Comment:
How about:

s"""Diplomacy Error: $context produced ${o.size} parameters for ${oPorts.size} outward ports. These sizes must match.
           | Outward parameters produced: [${o.mkString(",")}]
           | Outward ports connected: [${oPorts.map(_._2.name).mkString(",")}]
I am not totally clear on whether there is a meaningful distinction between "outer"/"outward"/"output" when talking about ports vs parameters here that it would be useful to make clear in this message. @terpstra ?